qcodes/instrument/channel.py

```diff
@@ -167,9 +140,10 @@ class ChannelList(Metadatable):
 
     """
 
-    def __init__(self, parent, name, chan_type: type, chan_list=None,
+    def __init__(self, parent: Instrument, name, chan_type: type, chan_list=None,
```

Comment:
Either we use type annotations or we don't (and I guess we are moving towards using them), but we should not mix. Also, the `__init__` function raises errors if `chan_type` and `multichan_paramclass` are not something which is more specific than `type`.